brigade-generic-gateway/cmd/brigade-generic-gateway/server_test.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/Azure/brigade/pkg/storage/mock"
+)
+
+func TestNewRouter(t *testing.T) {
+	s := mock.New()
+	s.ProjectList[0].ID = "brigade-4625a05cf6914e556aa254cb2af234203744de2f"
+	s.ProjectList[0].Name = "deis/empty-testbed"
+	s.ProjectList[0].GenericWebhookSecret = "mysecret"
+	r := newRouter(s)
+
+	if r == nil {
+		t.Fail()
+	}
+
+	ts := httptest.NewServer(r)
+	defer ts.Close()
+
+	res, err := http.Get(ts.URL + "/healthz")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if res.StatusCode != 200 {
+		t.Fatalf("Unexpected status on healthz: %s", res.Status)
+	}
+
+	body, err := ioutil.ReadFile("./testdata/genericwebhook.json")
+	if err != nil {
+		t.Fatal(err)
+	}
+
```

Comment:
I had this originally but it would have to hit an actual GitHub endpoint (haven't interface'd this behavior...). Would this be a proper test in this way?

Reply:
Ah, I see.  No, I think we're good -- especially in the happy path case coverage in `pkg`